backend/user-service/src/main/java/hu/me/iit/malus/thesis/user/service/impl/UserServiceImpl.java

```diff
@@ -335,62 +270,62 @@ public Teacher getTeacherByEmail(String teacherEmail) {
      * {@inheritDoc}
      */
     @Override
-    public Teacher getTeacherByCreatedCourseId(Long courseId) {
+    public TeacherDto getTeacherByCreatedCourseId(Long courseId) {
         try {
             Optional<Teacher> optTeacher = teacherRepository.findByCreatedCourseId(courseId);
-            if (!optTeacher.isPresent()) {
+            if (optTeacher.isEmpty()) {
                 throw new UserNotFoundException();
             }
-            return optTeacher.get();
+            return Converter.createTeacherDtoFromTeacher(optTeacher.get());
         } catch (DataAccessException e) {
             throw new DatabaseOperationFailedException(e);
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Set<Long> getRelatedCourseIds(String userEmail) {
+        Set<Long> relatedCourseIds = new HashSet<>();
+        User user = getAnyUserByEmail(userEmail);
+
+        if (user instanceof Student) {
```

Comment:
amikor castolni kell, az mindig valami tervezesi problema. pl. be lehetne vezetni egy getRelatedCorseIds a parentbe, a leszarmazottban pedig implementalod. beszeljunk rola.

Reply:
okés beszélünk